dg/media/app_dashboards/js/loop_dashboard.js

```diff
@@ -2925,37 +3365,297 @@ function aggregator_payment_sheet(data_json, aggregator) {
+            defaultContent: " "
+        }, {
+            title: "Farmer Comment",
+            defaultContent: " "
         }],
         "dom": 'T<"clear">rtip',
-        "pageLength": 2,
+        "pageLength": 10,
         "oTableTools": {
             "sSwfPath": "/media/social_website/scripts/libs/tabletools_media/swf/copy_csv_xls_pdf.swf",
             "aButtons": [{
-                "sExtends": "csv",
-                "sButtonText": "Download",
-                "bBomInc": true,
-                "sTitle": "Loop_India_Bihar_Aggregator Payment_" + getFormattedDate(aggregator) + "Payment Summary"
+
+                "sExtends": "text",
+                "sButtonText": "Edit",
+                "fnClick": function(nButton, oConfig) {
+                    $('#ToolTables_table2_1').removeClass('disable-button');
+                    editTable2 = 1;
+                    $('#table2').find('tr td:nth-child(5)').css('background-color', '#E5FED6');
+                    $('#table2').find('tr td:nth-child(7)').css('background-color', '#E5FED6');
+                }
+            }, {
+                "sExtends": "ajax",
+                "sButtonText": "Submit",
+                "sButtonClass": "disable-button",
+                "sAjaxUrl": "http://localhost:4001/loop/api/v1/aggregatorshareoutliers/",
+
+                "fnClick": function(nButton, oConfig) {
+                    var finalData = [];
+                    var finalDataFarmer = [];
+                    $('#table2').find('th').removeAttr("style");
+                    $('#table2').find('tr').removeAttr("style");
+                    $('#table2').find('td').removeAttr("style");
+                    editTable2 = 0;
+                    $('#ToolTables_table2_1').addClass('disable-button');
+                    finalData = processAggregatorRow(rows_table2,finalData);
+                    finalDataFarmer = processFarmerRow(rows_table2_farmer,finalDataFarmer);
+                    var sData = this.fnGetTableData(oConfig);
+                    var sData = this.fnGetTableData(oConfig);
+                    var aggregatorObjects = {
+                        "objects": finalData
+                    };
+                    var farmerObjects = {
+                        "objects": finalDataFarmer
+                    };
+                    
+                    if (Object.keys(rows_table2).length > 0) {
+                        $.ajax({
+                            url: oConfig.sAjaxUrl,
+                            type: 'patch',
+                            dataType: 'json',
+                            contentType: "application/json; charset=utf-8",
+                            headers:{"Authorization":"ApiKey "+window.localStorage.name+":"+window.localStorage.akey},
+                            data: JSON.stringify(aggregatorObjects),
+                            success: function() {
+                                alert("success");
+                                for (var keys in rows_table2) {
```

Comment:
why do we need to change the values again when already done on modal submit press.